pytorch_lightning/loggers/file_logger.py

```diff
@@ -0,0 +1,180 @@
+
+    def save(self):
+        hparams_file = os.path.join(self.log_dir, self.NAME_HPARAMS_FILE)
+        save_hparams_to_yaml(hparams_file, self.hparams)
+
+        if self.metrics:
+            with io.open(self.metrics_file_path, 'w', newline='') as f:
+                self.writer = csv.DictWriter(f, fieldnames=self.metrics_keys)
+                self.writer.writeheader()
+                self.writer.writerows(self.metrics)
+
+
+class FileLogger(LightningLoggerBase):
+    r"""
+    Log to local file system in yaml and CSV format. Logs are saved to
+    ``os.path.join(save_dir, name, version)``.
+
+    Example:
+        >>> from pytorch_lightning import Trainer
+        >>> from pytorch_lightning.loggers import FileLogger
+        >>> logger = FileLogger("logs", name="my_exp_name")
+        >>> trainer = Trainer(logger=logger)
+
+    Args:
+        save_dir: Save directory
+        name: Experiment name. Defaults to ``'default'``.
+        version: Experiment version. If version is not specified the logger inspects the save
+            directory for existing versions, then automatically assigns the next available version.
+    """
+
+    def __init__(self,
+                 save_dir: str,
+                 name: Optional[str] = "default",
+                 version: Optional[Union[int, str]] = None):
+
+        super().__init__()
+        self._save_dir = save_dir
+        self._name = name or ''
+        self._version = version
+        self._experiment = None
+
+    @property
+    def root_dir(self) -> str:
+        """
+        Parent directory for all checkpoint subdirectories.
+        If the experiment name parameter is ``None`` or the empty string, no experiment subdirectory is used
+        and the checkpoint will be saved in "save_dir/version_dir"
+        """
+        if self.name is None or len(self.name) == 0:
+            return self._save_dir
+        return os.path.join(self._save_dir, self.name)
+
+    @property
+    def log_dir(self) -> str:
+        """
+        The log directory for this run. By default, it is named
+        ``'version_${self.version}'`` but it can be overridden by passing a string value
+        for the constructor's version parameter instead of ``None`` or an int.
+        """
+        # create a pseudo standard path ala test-tube
+        version = self.version if isinstance(self.version, str) else f"version_{self.version}"
+        log_dir = os.path.join(self.root_dir, version)
+        return log_dir
+
+    @property
+    def experiment(self) -> ExperimentWriter:
+        r"""
+
+        Actual ExperimentWriter object. To use ExperimentWriter features in your
+        :class:`~pytorch_lightning.core.lightning.LightningModule` do the following.
+
+        Example::
+
+            self.logger.experiment.some_experiment_writer_function()
+
+        """
+        if self._experiment is not None:
+            return self._experiment
+
+        os.makedirs(self.root_dir, exist_ok=True)
```

Comment:
maybe we should warn the user here that potential logged values will be overwritten?

Reply:
In that case, should the exception be raised or just warn the user and overwrite existing files?

Reply:
Well, I left this line unchanged. This way it is consistent with the `TensorboardLogger`, see [here](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/pytorch_lightning/loggers/tensorboard.py#L106). If the experiment `root_dir` folder exists, it's content will not be deleted. 

The `root_dir` folder contains subfolders with experiment runs (versions), where the actual metrics and hparams are saved. 
Directory structure:
```
root_dir/
  version_1/
    hparams.yaml
    metrics.csv
  version_2/
  ...
```

If the user does not select the experiment version, next available version is chosen - a new subfolder is created. However, if the user selects an already existing version, the content will be overwritten. Therefore I added a warning to `ExperimentWriter` for that particular case, see [line](https://github.com/PyTorchLightning/pytorch-lightning/blob/c6fc5d3fc43d00d2b75940d1b94b129420778c07/pytorch_lightning/loggers/csv.py#L44).